test/functional/test_nvme_hotplug.py

```diff
@@ -0,0 +1,281 @@
+        conf = {}
+
+    @classmethod
+    def setUpClass(cls):
+        cls.start_node()
+
+    @classmethod
+    def tearDownClass(cls):
+        if conf:
+            cls.stop_node()
+
+    def test_step1_nvmedrive_remove(self):
+        ssh = sshclient.SSH("127.0.0.1", "root", "root", port=2222)
+        ssh.connect()
+        s = UnixSocket(path)
+        s.connect()
+        s.recv()
+
+        # 1.1: remove drive
+        print("------------Remove NVME device------------")
+        status, stdout = ssh.exec_command('nvme list')
+        print(stdout)
+        payload_enable_qmp = {
+            "execute": "qmp_capabilities"
+        }
+        s.send(json.dumps(payload_enable_qmp))
+        s.recv()
+
+        payload_drive_remove = {
+                "execute": "human-monitor-command",
+                "arguments": {
+                 "command-line": "device_del dev-nvme-1"
+                }
+        }
+        s.send(json.dumps(payload_drive_remove))
+        lines = s.recv()
+        print(lines)
+        s.close()
+        # around 10s is necessary for refresh the device list
+        time.sleep(10)
+        status, stdout = ssh.exec_command('nvme list')
+        self.assertNotIn('0400001C6BB4', stdout, "NVME drive remove fail")
+
+    def test_step2_nvmedrive_insert(self):
+        ssh = sshclient.SSH("127.0.0.1", "root", "root", port=2222)
+        ssh.connect()
+        s = UnixSocket(path)
+        s.connect()
+        s.recv()
+        status, stdout = ssh.exec_command('nvme list')
+        print(stdout)
+
+        # 2.1: insert known NVME drive
+        payload_enable_qmp = {
+            "execute": "qmp_capabilities"
+        }
+        s.send(json.dumps(payload_enable_qmp))
+        s.recv()
+
+        print("------------Insert NVME device------------")
+        payload_drive_insert = {
+                "execute": "human-monitor-command",
+                "arguments": {
+                    "command-line": "device_add nvme,"
+                                    "id=dev-nvme-1,drive=nvme-1,"
+                                    "model_number=INTEL-SSD00000001,"
+                                    "serial=FUKD72220009375A01,"
+                                    "bus=downstream4,cmb_size_mb=1"
+                }
+        }
+        s.send(json.dumps(payload_drive_insert))
+        lines = s.recv()
+        print(lines)
+        time.sleep(5)
+        status, stdout = ssh.exec_command('nvme list')
+        print(stdout)
+        self.assertIn('SSD00000001', stdout, "NVME drive insert fail")
+
+        # 2.2: IO test, nvme1n1, SSD00000001
+        status, stdout = ssh.exec_command(
+            'sudo fio -filename=/dev/nvme1n1  -direct=1 -iodepth 1 -thread \
+             -rw=write -ioengine=psync -bs=4k -size=10M -numjobs=10 \
+             -runtime=100 -do_verify=1 -group_reporting -name=mytest')
+        time.sleep(5)
+        self.assertNotIn('error', stdout, "New NVME drive r/w test fail")
+        print(stdout)
```

Comment:
recommend to remove it.
if last assert failed, this print will not be called.
if last assert succeed, this print is not cared.

Reply:
Hi Aaron,I already fixed these issues according your advises. Please help review it. Thanks.